src/history/hs.c

```diff
@@ -279,6 +279,12 @@ __wt_hs_cursor(WT_SESSION_IMPL *session, WT_CURSOR **cursorp, uint32_t *session_
                 if (!cache->hs_session_inuse[i]) {
                     *cursorp = cache->hs_session[i]->hs_cursor;
                     cache->hs_session_inuse[i] = true;
+                    if (reader) {
+                        cache->hs_session[i]->iface.begin_transaction(
```

Comment:
Since we're writing updates to the lookaside with timestamps, I think we need to set the read timestamp when finding lookaside updates. This explains why we're failing to find lookaside updates even in very obvious situations.

Reply:
Updates in history store has the same timestamp of the original update. It should be visible to user reads.

Reply:
Isn't this to avoid reading a tombstone when you shouldn't?

Reply:
> Updates in history store have the same timestamp of the original updates. They should be visible to user reads.

But setting the timestamp is something you do to a session. And history store cursors have their own dedicated sessions. Therefore you have different visibility I think.

Does this seem right @kommiharibabu?